src/client.cpp

```diff
@@ -112,16 +117,20 @@ std::vector<Json::Value> waybar::Client::getOutputConfigs(struct waybar_output &
   return configs;
 }
 
-void waybar::Client::handleOutputName(void *      data, struct zxdg_output_v1 * /*xdg_output*/,
-                                      const char *name) {
+void waybar::Client::handleOutputDescription(void *data, struct zxdg_output_v1 * /*xdg_output*/,
+                                             const char *description) {
   auto client = waybar::Client::inst();
   try {
-    auto &output = client->getOutput(data);
-    output.name = name;
-    spdlog::debug("Output detected: {} ({} {})",
-                  name,
-                  output.monitor->get_manufacturer(),
-                  output.monitor->get_model());
+    auto &      output = client->getOutput(data);
+    const char *open_paren = strrchr(description, '(');
+    const char *close_paren = strrchr(description, ')');
+
+    // Description format: "identifier (name)"
+    size_t identifier_length = open_paren - description;
+    output.identifier = std::string(description, identifier_length - 1);
+    output.name = std::string(description + identifier_length + 1, close_paren - open_paren - 1);
```

Comment:
Quote from xdg-output-unstable-v1:
```
Many compositors can produce human-readable descriptions of their
outputs.  The client may wish to know this description as well, to
communicate the user for various purposes.

The description is a UTF-8 string with no convention defined for its
contents. Examples might include 'Foocorp 11" Display' or 'Virtual X11
output via :1'.
```
It's not correct to rely on any specific content or format of description string.

If you need to handle both `name` and `description` you'll have to define `name` and `description` callbacks that only set corresponding properties, subscribe to `zxdg_output_v1.done` or  `wl_output.done` and wait for the `done` callback before taking any actions. The callback should be selected based on zxdg_output_v1 version; anything >= 3 should use `wl_output.done`, previous versions should use `zxdg_output_v1.done`.

Reply:
> The callback should be selected based on zxdg_output_v1 version;

I checked the code and we are binding to version 2 explicitly. You can ignore that part and use only `zxdg_output_v1.done`.

Reply:
I addressed your comments. The worst that can happen now is that the identifier is some garbage in case the description is not in the format we want. Though I feel like this is acceptable as swaybar does this? :)